dll/win32/browseui/shellfind/CFindFolder.cpp

```diff
@@ -271,8 +301,9 @@ static UINT RecursiveFind(LPCWSTR lpPath, _SearchData *pSearchData)
 
             uTotalFound += RecursiveFind(szPath, pSearchData);
         }
-        else if ((pSearchData->szFileName.IsEmpty() || PathMatchSpecW(FindData.cFileName, pSearchData->szFileName))
-                && (pSearchData->szQueryA.IsEmpty() || SearchFile(szPath, pSearchData)))
+        else if (FileNameMatch(FindData.cFileName, pSearchData)
+                && QueryMatch(szPath, pSearchData)
+                && AttribHiddenMatch(FindData.dwFileAttributes, pSearchData))
```

Comment:
The `AttribHiddenMatch` check should come first, because `QueryMatch()` is an expensive operation that needs to go through the file's contents.

Reply:
Changed.